### PR TITLE
[infra] upgrade the CRA to the latest and set up ESLinting rules

### PR DESCRIPTION
**Pull request by anton-karlovskiy**
- Upgrade the CRA packages to the latest and add features from the latest CRA.
  * Remove some packages (mostly related to testing in case they are not needed).
  * Omit `yarn test` from husky hooks for now due to ^(above).
- Add ESLinting rules and fix linting errors/warnings.

### CODE REVIEW

**Comment by anton-karlovskiy** [comment]
@nud3l 
I updated the CRA to the latest because I run into a tricky issue when trying to set up ESLinting rules in the project.
I think that issue was caused by some dependency conflicts or something and according to GitHub threads, that issue would occur with `react-script` 3.4.1 (the original version).
Re:
https://github.com/eslint/eslint/issues/13283
https://stackoverflow.com/questions/60431931/eslint-error-eslint-couldnt-find-the-config-react-app

Considering that we should update/clean-up the test packages, I updated the CRA and cleaned up some unnecessary packages.
Re:
https://docs.google.com/document/d/1DZZbKPOcguTxeTJY4CTDk3pVT9EW_Lx_qsXgn992vmE/edit?disco=AAAALjMgVJ4
https://docs.google.com/document/d/1DZZbKPOcguTxeTJY4CTDk3pVT9EW_Lx_qsXgn992vmE/edit?disco=AAAALjMgVJ8

**Comment by anton-karlovskiy** [comment]
@nud3l cc @alexeiZamyatin 
### dependency differences between after/before package.json
dependencies-diff

### devDependency differences between after/before package.json
devDependencies-diff

### rest differences between after/before package.json
rest-diff

As you can see, I cleaned up some packages for testing in devDependencies because those packages will not be used for React testing now.
So I'd suggest not running `yarn test` when committing but I'd like your opinions first.

**Comment by anton-karlovskiy** [comment]
@nud3l cc @alexeiZamyatin 
The rest of the changes are totally for fixing ESLinting errors/warnings so I believe they will not break the logic.
I'd say we should write code on top of these JavaScript standards (mostly following React standard rules and Google standard rules).

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    ],
+    'quote-props': [
+      'error',
+      'as-needed'
+    ],
+    'capitalized-comments': 'off',
+    'max-len': [
+      'warn',
+      {
+        code: 130
```
Comment (nud3l) [inline .eslintrc.js:54]: We use 120 across our repos

Comment (anton-karlovskiy) [reply .eslintrc.js:54]: OK, then I will update it to 120. :)

Comment (anton-karlovskiy) [reply .eslintrc.js]: Done!

**Comment by nud3l** [comment]
Please fix lint warnings
```
yarn run v1.22.5
$ react-app-rewired build
Creating an optimized production build...
Compiled with warnings.

src/App.tsx
  Line 158:9:  Unexpected console statement  no-console

src/common/components/dashboard-table/dashboard-table.tsx
  Line 28:5:  Using target="_blank" without rel="noreferrer" is a security risk: see https://html.spec.whatwg.org/multipage/links.html#link-type-noopener  react/jsx-no-target-blank

src/common/components/oracle-table/oracle-table.tsx
  Line 45:9:  Unexpected console statement  no-console

src/common/components/topbar.tsx
  Line 71:7:  Unexpected console statement  no-console

src/common/components/vault-table/vault-table.tsx
  Line 38:9:   Unexpected console statement  no-console
  Line 90:11:  Unexpected console statement  no-console

src/common/live-data/balances-watcher.ts
  Line 27:5:  Unexpected console statement  no-console

src/common/live-data/block-height-watcher.ts
  Line 19:5:  Unexpected console statement  no-console

src/common/live-data/issue-transaction.watcher.ts
  Line 48:5:  Unexpected console statement  no-console

src/common/live-data/price-watcher.ts
  Line 24:9:  Unexpected console statement  no-console

src/common/live-data/redeem-transaction.watcher.ts
  Line 37:5:  Unexpected console statement  no-console

src/common/live-data/totals-watcher.ts
  Line 23:5:  Unexpected console statement  no-console

src/pages/app/issue/enter-btc-amount.tsx
  Line 63:9:   Unexpected console statement  no-console
  Line 141:7:  Unexpected console statement  no-console

src/pages/app/issue/modal/payment-view.tsx
  Line 37:9:  Unexpected console statement  no-console

src/pages/app/redeem/enter-amount-and-address.tsx
  Line 69:9:  Unexpected console statement  no-console
  Line 84:9:  Unexpected console statement  no-console

src/pages/app/redeem/modal/reimburse-view.tsx
  Line 41:9:  Unexpected console statement  no-console
  Line 60:7:  Unexpected console statement  no-console
  Line 76:7:  Unexpected console statement  no-console

src/pages/app/redeem/modal/status-view.tsx
  Line 42:9:  Unexpected console statement  no-console

src/pages/app/transfer/transfer.tsx
  Line 39:7:  Unexpected console statement  no-console

src/pages/dashboard/components/active-staked-relayers.tsx
  Line 12:7:  'ActiveStakedRelayers' is already defined  @typescript-eslint/no-redeclare

src/pages/dashboard/components/btc-relay.tsx
  Line 57:9:  Unexpected console statement  no-console

src/pages/dashboard/components/collateralization.tsx
  Line 35:9:  Unexpected console statement  no-console

src/pages/dashboard/components/oracle-status.tsx
  Line 44:9:  Unexpected console statement  no-console

src/pages/dashboard/components/parachain-security.tsx
  Line 41:9:  Unexpected console statement  no-console

src/pages/dashboard/issue/issue.dashboard.page.tsx
  Line 121:7:  Unexpected console statement  no-console
  Line 130:7:  Unexpected console statement  no-console

src/pages/dashboard/parachain/parachain.dashboard.page.tsx
  Line 55:7:  Unexpected console statement  no-console
  Line 63:7:  Unexpected console statement  no-console

src/pages/dashboard/redeem/redeem.dashboard.page.tsx
  Line 105:7:  Unexpected console statement  no-console
  Line 140:9:  Unexpected console statement  no-console

src/pages/dashboard/relay/relay.dashboard.page.tsx
  Line 35:9:  Unexpected console statement  no-console

src/pages/vault-dashboard/issue-table/issue-table.tsx
  Line 32:9:  Unexpected console statement  no-console

src/pages/vault-dashboard/redeem-table/redeem-table.tsx
  Line 31:9:  Unexpected console statement  no-console

src/pages/vault-dashboard/replace-table/replace-table.tsx
  Line 37:9:  Unexpected console statement  no-console

src/pages/vault-dashboard/update-collateral/update-collateral.tsx
  Line 130:7:  Unexpected console statement  no-console
```

**Comment by theodorebugnet** [comment]
We've been using 4-space indent everywhere so far, is it really worth switching the entire project to 2-space indent? @nud3l

**Comment by anton-karlovskiy** [comment]
@theodorebugnet 
Yes, I understand this concern.

As far as I know, actually, now almost all standards recommend 2 space indents.
And I'd say it's helpful to see the code as the number of characters in a line can be small.

But I'm open to listening. cc @nud3l

**Review by nud3l — APPROVED** [review]

**Pull request merged**